Added Native Touch Support

lelandcope opened the pull request:
Ive ported over code Ive written so there can be native touch support without the use of hammertime.js This has been tested on the ipad and iphone.

voronianski commented on js/ngDialog.js:
```diff
@@ -5,16 +5,16 @@
+	} else {
+		// Global Variables
+		factory(root.angular);
+	}
```
@lelandcope that is bad, indentation shouldn't be changed - see [CONTRIBUTING](https://github.com/likeastore/ngDialog/blob/master/CONTRIBUTING.md).

lelandcope replied:
Its funny but I had them as spaces and changed them because they didn't match the style in the file. I will fix that. I didnt see the contributing file. Ill make sure it matches what was set in that file.

voronianski commented on js/ngDialog.js:
```diff
@@ -527,7 +527,54 @@
+								var timeThreshold	 = 500;
+								var tapped			= false;
+
+								$dialog.on("touchstart", function(startEvent) {
```
@lelandcope please code style issues, see [CONTRIBUTING](https://github.com/likeastore/ngDialog/blob/master/CONTRIBUTING.md).

voronianski commented on js/ngDialog.js:
```diff
@@ -527,7 +527,54 @@
+										}
+									};
+									
+									moveHandler = function(moveEvent) {
```
@lelandcope I assume that all these handlers can be stored in private methods, without recreating them on every event.

lelandcope commented:
I went and cleaned up the code to match the style in the contrib file. Because of how tap detection needs to work Im not able to the handlers to the private functions section without adding a bunch of extra code which didnt make sense to do just so those functions could be in the private functions object. I have cleaned the code up though which should make it much easier to follow. Let me know if there is more you want me to change.

curts commented:
I am not able to tell what changed in this PR - all I see are white space changes...?? E.g.:

![image](https://cloud.githubusercontent.com/assets/1950836/6743397/6625ecfc-ce67-11e4-9b68-431158fa5142.png)

lelandcope commented:
Ill re-fork and then send you a new pull request without all the white spaces that way you see just the changes.

mansehr commented:
Any updates on this @lelandcope ?
